sdk/client.go

```diff
@@ -83,3 +85,19 @@ func (c *Client) getServiceStatus(resPath string) (*api.MetaMessage, error) {
 	}
 	return msg, nil
 }
+
+func (c *Client) postServiceAction(resPath string, req interface{}) error {
+	resp, resErr := c.postResource(resPath, req)
+	// Since the response body can have more information about the error, we try to parse it
+	if resErr != nil && resp == nil {
+		return fmt.Errorf("postResource: %w", resErr)
+	}
+	msg := api.MetaMessage{}
+
+	if err := json.Unmarshal(resp, &msg); err != nil {
```

Comment:
i'd rely on http StatusCodes for this vs serialization success/failure and not try and overcomplicate it. Someone could return a 500 from api, but implement a MetaMessage response successfully and then this will be trouble

Reply:
Ok I pushed a proposed fix for it. 
Now it should not happen, in any case when an error is returned from the `endpoint` SDK follows that if status ok is returned it returns nil.